src/AlfangoDB/modules/read.mo

```diff
@@ -179,4 +182,91 @@ module {
         Prelude.unreachable();
     };
 
+    public func getDatabaseInfo(
+        alfangoDB: Database.AlfangoDB
+    ) : OutputTypes.GetDatabasesOutputType {
+
+        // create an empty array
+        var databasesInfo : [
+            {
+                name: Text;
+                tables: [Text];
+            }
+        ] = [];
+
+        // crete an Iterator
+        let databaseArgsIter = Iter.range(0, alfangoDB.databases.size() - 1);
```

Comment:
1. Do not use nested switch cases, instead use `?do{}` (option block)
2. Instead of Array.append() use Buffer https://internetcomputer.org/docs/current/motoko/main/base/Buffer

Reply:
Refer src/AlfangoDB/modules/read.mo to understand usage of _option block_ and _Buffer_.